### PR TITLE
Android fixes

### PR DESCRIPTION
**Pull request by angel-rs**

### CODE REVIEW

File: android/src/main/java/com/vydia/RNUploader/UploaderModule.kt [hunk 1]
```diff
@@ -318,6 +318,16 @@ class UploaderModule(val reactContext: ReactApplicationContext) : ReactContextBa
...
+  @ReactMethod
+  fun addListener(type: String?) {
+      // Keep: Required for RN built in Event Emitter Calls.
+  }
+
+  @ReactMethod
+  fun removeListeners(type: Int?) {
+      // Keep: Required for RN built in Event Emitter Calls.
+  }
+
```
Comment (angel-rs) [inline android/src/main/java/com/vydia/RNUploader/UploaderModule.kt:330]: Fixes this warning:
https://stackoverflow.com/questions/69538962/new-nativeeventemitter-was-called-with-a-non-null-argument-without-the-requir

File: android/src/main/java/com/vydia/RNUploader/UploaderReactPackage.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class UploaderReactPackage : ReactPackage {
+    override fun createViewManagers(
+        reactContext: ReactApplicationContext
+    ): MutableList<ViewManager<View, ReactShadowNode<*>>> = mutableListOf()
+
+    override fun createNativeModules(
+        reactContext: ReactApplicationContext
+    ): MutableList<NativeModule> = listOf(UploaderModule(reactContext)).toMutableList()
+}
```
Comment (angel-rs) [inline android/src/main/java/com/vydia/RNUploader/UploaderReactPackage.kt:18]: Replaced the java version with kotlin, following the RN's Android native modules docs
https://reactnative.dev/docs/native-modules-android?android-language=kotlin

File: example/RNBackgroundExample/package.json [hunk 3]
```diff
@@ -18,8 +18,8 @@
     "e2e/test/android": "detox test -c android"
   },
   "dependencies": {
-    "react": "18.1.0",
-    "react-native": "0.70.2",
+    "react": "18.2.0",
+    "react-native": "0.71.7",
```
Comment (angel-rs) [inline example/RNBackgroundExample/package.json:22]: Updated to use same version as our app

File: android/build.gradle [hunk 4]
```diff
@@ -24,7 +24,6 @@ buildscript {
 
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
```
Comment (angel-rs) [inline android/build.gradle:27]: Removing this fixes this deprecation warning
https://stackoverflow.com/questions/65179275/the-kotlin-android-extensions-gradle-plugin-is-deprecated

**Comment by angel-rs** [comment]
Note: tried getting the example app working, but wasn't able to.
Ideally we should try to get it working, alongside the original repo detox e2e tests.

**Review by timjbray — APPROVED** [review]

File: android/src/main/java/com/vydia/RNUploader/GlobalRequestObserverDelegate.kt [hunk 5]
```diff
@@ -22,16 +24,21 @@ class GlobalRequestObserverDelegate(reactContext: ReactApplicationContext) : Req
...
-    } else {
-      params.putString("error", "Unknown exception")
+    when (exception) {
+      is UserCancelledUploadException -> {
+        errorMessage = "User cancelled upload"
+      }
+      is UploadError -> {
+        errorMessage = "responseCode=${exception.serverResponse.code} responseBody=${exception.serverResponse.bodyString}"
+        Log.e(TAG, "Error, upload error: ${exception.serverResponse.code} ${exception.serverResponse.bodyString}")
+      }
```
Comment (angel-rs) [inline android/src/main/java/com/vydia/RNUploader/GlobalRequestObserverDelegate.kt:38]: This ensures we forward the S3 upload error to the js side, which fixes upload retry if the presigned url is expired.
Will also help know the reason of failure to upload as before we just got a `Upload error`

**Pull request merged**